### PR TITLE
Added three ESDIRK schemes [RK_solvers]

### PR DESCRIPTION
**Pull request by bensworth**
I added the implicit trapezoid rule, L-stable ESDIRK-32, and A-stable ESDIRK-33. The latter two are stiffly accurate, and all three have stage-order two. 

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#964](https://github.com/mfem/mfem/pull/964) | @bensworth | @tzanio | @jandrej + @cjvogl + @vladotomov | 10/10/20 | 10/29/20 | 11/03/20 | |

### CODE REVIEW

**Comment by bensworth** [comment]
@tzanio what is the deal with this? It seems to pass all the tests. Are there other things I need to do?

**Comment by tzanio** [comment]
@bensworth, when it is ready you are also supposed to add the `ready-for-review` label :)

The bot is just reminding us that the PR is old and has not been processed, which is a good thing...

**Comment by vladotomov** [comment]
@bensworth could you run these through the convergence tests, see `tests/unit/linalg/test_ode2.cpp`.

**Comment by bensworth** [comment]
> @bensworth could you run these through the convergence tests, see `tests/unit/linalg/test_ode2.cpp`.

@vladotomov What's the correct way to do this? I added the new integrators to `test_ode.cpp` in `/tests/unit/linalg` directory, but the linalg directory doesn't seem to copy to my build directory. In `my_build/tests/unit/' the only executables I see  `psedov_tests_cpu    punit_tests      sedov_tests_debug psedov_tests_debug  sedov_tests_cpu  unit_tests`. Note, when I run `make -j test ` in my build directory, I pass all but one test:
```
99% tests passed, 1 tests failed out of 84

Total Test time (real) = 118.06 sec

The following tests FAILED:
	  4 - punit_tests (Failed)
Errors while running CTest
```
This happens with or without the new integrators though.

**Comment by jandrej** [comment]
It should be encapsulated in the unit_tests executable. Your tests belong into `test_ode.cpp` not the second order ODE integrator test.

Depending on what build system you use (with cmake you run `make unit_tests` in your build directory). Then you can run
```
./tests/unit/unit_tests -s
```
and `grep` for your test tag, or just run the tag separately.

**Comment by bensworth** [comment]
Thanks @jandrej, added unit tests and they all pass.

**Comment by jandrej** [comment]
Travis seems not happy with the code formatting. Can you run `make style` with the correct `astyle` version in your mfem root directory? If that's too much of a setup, let me know and I'll apply the style changes to the branch.

**Comment by bensworth** [comment]
I don't know about the make style/astyle? I did a quick look at travis and think the issue was I used 4-space tabs, not three. If that doesn't fix it, if you could run the astyle that would be great.

**Review by cjvogl — APPROVED** [review]
Looks to good to me: passed `unit-tests` (didn't run autotest because it wouldn't cover this code)

**Review by vladotomov — APPROVED** [review]
Could be mentioned in the changelog.

**Comment by tzanio** [comment]
@jandrej, can you approve?

**Review by jandrej — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

File: linalg/ode.hpp [hunk 1]
```diff
@@ -431,6 +431,48 @@ class SDIRK33Solver : public ODESolver
...
+public:
+   virtual void Init(TimeDependentOperator &_f);
+
+   virtual void Step(Vector &x, double &t, double &dt);
+};
+
+
+/** Three stage, explicit singly diagonal implicit Runge-Kutta (ESDIRK) method
+    of order 2. L-stable. */
+class ESDIRK32Solver : public ODESolver
```
Comment (v-dobrev) [inline linalg/ode.hpp:450]: Just curious: how is a method "explicit" and "implicit" at the same time? 😄

Comment (bensworth) [reply linalg/ode.hpp:450]: It's a mutant hybrid method! Probably there should be an arrow, E->SDIRK schemes, one explicit stage followed by a set of SDIRK stages

**Comment by v-dobrev** [comment]
Merged in post v4.2 `next`.

**Comment by tzanio** [comment]
Should this be mentioned in `CHANGELOG`?

**Comment by tzanio** [comment]
Thanks for adding these Ben!

**Pull request merged**